src/mlpack/core/data/load_impl.hpp

```diff
@@ -410,6 +410,110 @@ bool LoadCategorical(const std::string& filename,
   return true;
 }
 
+template<typename MatType>
+bool Load(const std::vector<std::string>& filenames,
+          MatType& matrix,
+          const TextOptions& opts)
+{
+  TextOptions copyOpts(opts);
+  return Load(filenames, matrix, copyOpts);
+}
+
+template<typename MatType>
+bool Load(const std::vector<std::string>& filenames,
+          MatType& matrix,
+          TextOptions& opts)
+{
+  bool success = false;
+  MatType tmp;
+  arma::field<std::string> firstHeaders;
+  if (filenames.empty())
+  {
+    if (opts.Fatal())
+    {
+      Log::Fatal << "Load(): Please specify the filenames to be loaded!"
+          << std::endl;
+    }
+    else
+    {
+      Log::Warn << "Load(): Please specify the filenames to be loaded!"
+          << std::endl;
+      return false;
+    }
+  }
+  for (size_t i = 0; i < filenames.size(); ++i)
+  {
+    success = Load(filenames.at(i), matrix, opts);
+    if (opts.HasHeaders())
+    {
+      if (i == 0)
+        firstHeaders = opts.Headers();
+      else
+      {
+        arma::field<std::string>& headers = opts.Headers();
+        for (size_t j = 0; j < headers.size(); ++j)
+        {
+          if (firstHeaders.at(0) != headers.at(0))
+            Log::Warn << "Load(): Headers from different CSV does not match!"
+                << std::endl;
+        }
+      }
+    }
+
+    if (success)
+    {
+      if (i == 0)
+      {
+        tmp = std::move(matrix);
+      }
+      else
+      {
+        if (!opts.NoTranspose()) // if transpose
+        {
+          if (tmp.n_rows != matrix.n_rows)
+          {
+            if (opts.Fatal())
+              Log::Fatal << "Load(): dimension mismatch when loading vectors"
+                << std::endl;
+            else
+            {
+              Log::Warn << "Load(): dimension mismatch when loading vectors"
+                << std::endl;
+              return false;
+            }
+          }
+          else
+            tmp = join_rows(tmp, matrix);
+        }
+        else
+        {
+          if (tmp.n_cols != matrix.n_cols)
+          {
+            if (opts.Fatal())
+              Log::Fatal << "Load(): dimension mismatch when loading vectors"
+                << std::endl;
+            else
+            {
+              Log::Warn << "Load(): dimension mismatch when loading vectors"
+                << std::endl;
+              return false;
+            }
```

Comment:
Just as an aside, we have so many things of the form

```
if (fatal)
  Log::Fatal << "something";
else
  Log::Warn << "something";
  
return false;
```

I have to wonder if we could replace all these patterns with a utility function.  But, maybe another PR or cleanup.

Reply:
Yes, at the end wil be easier